app/models/spree/gateway/culqi_gateway.rb

```diff
@@ -0,0 +1,84 @@
+module Spree
+  class Gateway::CulqiGateway < Gateway
+    preference :public_key, :string
+    preference :secret_key, :string
+
+    def default_currency
+      "PEN"
+    end
+
+    if SolidusSupport.solidus_gem_version < Gem::Version.new('2.3.x')
+      def method_type
+        'culqi'
+      end
+    else
+      def partial_name
+        'culqi'
+      end
+    end
+
+    def provider_class
+      self.class
+    end
+
+    def authorize(amount, creditcard, gateway_options)
+      init_culqi
+      commit(amount, creditcard[:gateway_payment_profile_id], gateway_options, false)
+    end
+
+    def purchase(amount, creditcard, gateway_options)
+      init_culqi
+      commit(amount, creditcard[:gateway_payment_profile_id], gateway_options, true)
+    end
+
+    def capture(amount, response_code, gateway_options)
+      init_culqi
+      charge = Culqi::Charge.capture(response_code)
+      parse_response(charge)
+    end
+
+    def credit(amount, creditcard, gateway_options)
+      # Culqi only accepts 'duplicado','fraudulento' o 'solicitud_comprador'
+      # like reason's value
+      refund = Culqi::Refund.create(
+        :amount => amount,
+        :charge_id => creditcard,
+        :reason => "solicitud_comprador"
+      )
+      parse_response(refund)
+    end
+
+    def void(creditcard, gateway_options)
```

Comment:
I couldn't find a way to cancel an authorization in [culqi documentation](https://culqi.com/docs).
In docs said that authorizations that are not captured in 4 days expire automatically.
I plan to write them to confirm.

But, what should do for this method in case there is no way to cancel authorizations?